hamilflow/models/harmonic_oscillator/__init__.py

```diff
@@ -155,6 +147,7 @@ def __init__(
         system: Dict[str, float],
         initial_condition: Optional[Dict[str, float]] = {},
     ):
+        initial_condition = parse_ic_for_sho(system["omega"], **initial_condition)
```

Comment:
To be honest I am not happy with the syntax `system["omega"]`. It breaks the design and steals data from the `system` dictionary. I guess we need to improve the design at some point. @emptymalei